In [1]:
import numpy as np
import pandas as pd


### Урок 6. Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

#### Задание №1
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [3]:
# Ковариация
c_cov = np.mean(zp*ks) - np.mean(zp) * np.mean(ks)
c_cov

9157.839999999997

In [4]:
f_cov = np.cov(zp, ks, ddof=0)
f_cov

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [5]:
# коэффициент корреляции Пирсона
c_cov / (np.std(zp) * np.std(ks))

0.8874900920739158

In [6]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [7]:
df = pd.DataFrame({1:zp, 2: ks})
df.corr()

1        2
1  1.00000  0.88749
2  0.88749  1.00000

#### Задание №2
Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [8]:
iq = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]  
len(iq)

10

In [9]:
iq_mean = np.mean(iq)
iq_mean

118.1

In [10]:
iq_std = np.std(iq, ddof=1)
iq_std

10.54566788359614

In [11]:
#Найдем доверительный интервал
r_iq = iq_mean + 2.262 * iq_std / len(iq)**0.5
r_iq

125.64339223691834

In [12]:
l_iq = iq_mean - 2.262 * iq_std / len(iq)**0.5
l_iq

110.55660776308164

Доверительный интервал будет от 110.5 до 125.6

#### Задание №3
Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [13]:
# Доверительный интервал
r_h = 174.2 + 1.96 * 25**0.5 / 27**0.5
r_h

176.08601087935276

In [14]:
l_h = 174.2 - 1.96 * 25**0.5 / 27**0.5
l_h

172.31398912064722

Доверительный интервал будет от 172.31 до 176.08